src/main/resources/templates/academicCourseSection.html

```diff
@@ -56,7 +56,7 @@ <h1>New Academic course</h1>
                             th:field="*{subject}">
                         <option  th:value="*{subject}"></option>
                         <option  th:each="subject : ${subjects}" th:value="${subject.id}"
-                                th:text="${subject.getName()}"></option>
+                                 th:text="${subject.getName()}"></option>
                     </select>
                     <div class="error" th:if="${#fields.hasErrors('subject')}"
                          th:errors="*{subject}"></div>
```

Comment:
The error message “There is no new selection.” is not working for all possible cases while not selecting  any teacher. It works on and off.